File: BE/Laravel_Api/whatjob/app/Models/Employer.php
<?php

namespace App\Models;

use App\Helpers\RelationshipsTrait;
use App\Helpers\ValidatesTrait;
use Illuminate\Database\Eloquent\Concerns\HasUuids;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Employer extends Model
{
    use HasFactory, HasUuids;
    use RelationshipsTrait, ValidatesTrait;


    protected $fillable = [
        //'id',
        'companyName',
        'logo',
        'description',
        'hotline',
        'address',
    ];

    protected $rules = [
        // 'id' => 'required'
    ];

    public function user(){
        return $this->belongsTo(User::class, 'id', 'id');
    }

    public function recruitmentaricle(){
        return $this->hasMany(RecruitmentArticle::class);
    }
}

File: FE/whatjob/lib/model/employer.dart
class Employer {
  final String companyName;
  final String logo;
  final String description;
  final String hotLine;
  final String address;

  Employer({
    required this.companyName,
    required this.logo,
    required this.description,
    required this.hotLine,
    required this.address,
  });

  Employer.empty()
      : companyName = '',
        logo = '',
        description = '',
        hotLine = '',
        address = '';

  factory Employer.fromJson(Map<String, dynamic> json) {
    return Employer(
      companyName: json['companyName'] ?? '',
      logo: json['logo'] ?? '',
      description: json['description'] ?? '',
      hotLine: json['hotLine'] ?? '',
      address: json['address'] ?? '',
    );
  }

  @override
  String toString() {
    return '\nEmployer{'
        'companyName: $companyName, '
        'logo: $logo, '
        'description: $description, '
        'hotLine: $hotLine, '
        'address: $address'
        '}';
  }
}